File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/WizardsAndWarriors/Wizard.cs
namespace Aio_Algorithms_And_Solutions.Solutions.Exercism.CSharp.WizardsAndWarriors;

public class Wizard : Character
{
    private int _damage;

    public Wizard() : base("Wizard")
    {
        _damage = 3;
    }

    public override int DamagePoints(Character target)
    {
        return _damage;
    }

    public override bool Vulnerable()
    {
        return _damage == 3;
    }

    public void PrepareSpell()
    {
        _damage = 12;
    }    
}

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/WizardsAndWarriors/Wizard.java
package exercism;

public class Wizard extends Fighter {
    private boolean preparedSpell = false;

    @Override
    boolean isVulnerable() {
        return !preparedSpell;
    }

    @Override
    int damagePoints(Fighter warrior) {
        return preparedSpell ? 12 : 3;
    }

    void prepareSpell() {
        preparedSpell = true;
    }

}
